src/pudl/transform/ferc714.py

```diff
@@ -0,0 +1,749 @@
+from one year to the next, and these result in duplicate records, which are
+dropped.
+"""
+
+TZ_CODES = {
+    "EST": "America/New_York",
+    "EDT": "America/New_York",
+    "CST": "America/Chicago",
+    "CDT": "America/Chicago",
+    "MST": "America/Denver",
+    "MDT": "America/Denver",
+    "PST": "America/Los_Angeles",
+    "PDT": "America/Los_Angeles",
+    "AKST": "America/Anchorage",
+    "AKDT": "America/Anchorage",
+    "HST": "Pacific/Honolulu",
+}
+"""Mapping between standardized time offset codes and canonical timezones."""
+
+MISSING_UTILITY_ID_EIA = {
+    307: 14354,  # PacifiCorp
+    295: 40229,  # Old Dominion Electric Cooperative
+    329: 39347,  # East Texas Electricity Cooperative
+}
+"""FERC respondents lacking EIA utility IDs that appear to exist."""
+
+RENAME_COLS = {
+    "respondent_id_ferc714": {
+        "respondent_id": "utility_id_ferc714",
+        "respondent_name": "utility_name_ferc714",
+        "eia_code": "utility_id_eia",
+    },
+    "pa_demand_hourly_ferc714": {
+        "report_yr": "report_year",
+        "plan_date": "report_date",
+        "respondent_id": "utility_id_ferc714",
+        "timezone": "utc_offset_code",
+    },
+    "pa_description_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+        "elec_util_name": "utility_name_ferc714",
+        "peak_summer": "peak_demand_summer_mw",
+        "peak_winter": "peak_demand_winter_mw",
+    },
+    "id_certification_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "ba_gen_plants_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "ba_demand_monthly_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "ba_net_energy_load_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "adjacent_bas_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "ba_interchange_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "ba_lambda_hourly_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "lambda_description_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+    "pa_demand_forecast_ferc714": {
+        "report_yr": "report_year",
+        "respondent_id": "utility_id_ferc714",
+    },
+}
+
+##############################################################################
+# Internal helper functions.
+##############################################################################
+
+
+def _hours_to_ints(col_name):
+    """A macro to rename hourly demand columns."""
+    if re.match(r"^hour\d\d$", col_name):
```

Comment:
@zaneselvans you have told me to be wary of having multiple points of returns in a function. this one is very simple, so it is not a big deal, but you could certainly just rename `col_name` in  the `if` and without an `else` return col_name

Reply:
Fair enough. Changed.